pkg/secrethub/acl.go

```diff
@@ -296,18 +296,17 @@ func (c *Client) getAccessLevel(path api.BlindNamePath, accountName api.AccountN
 // Iterator returns an iterator that retrieves all access rules that apply to a
 // directory.
 func (s accessRuleService) Iterator(path string, params *AccessRuleIteratorParams) AccessRuleIterator {
-	var depth int
-	var ancestors bool
 	if params == nil {
-		depth = -1
-		ancestors = false
-	} else if params.Depth == nil {
-		depth = -1
-		ancestors = params.Ancestors
-	} else {
+		params = &AccessRuleIteratorParams{}
+	}
+
+	var depth int
+	if params.Depth != nil {
 		depth = int(*params.Depth)
-		ancestors = params.Ancestors
+	} else {
+		depth = -1
```

Comment:
We can even go one step further and do:

```Go
depth := -1
if params.Depth != nil {
  depth = int(*params.Depth)
}
```

Which do you prefer?

Reply:
I prefer the shortest version